okhttp-tests/src/test/java/okhttp3/internal/http/RecordingProxySelector.java

```diff
@@ -45,9 +45,8 @@ public void assertRequests(URI... expectedUris) {
 
   @Override public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {
     InetSocketAddress socketAddress = (InetSocketAddress) sa;
-    failures.add(
-        Util.format("%s %s:%d %s", uri, socketAddress.getHostName(), socketAddress.getPort(),
```

Comment:
calling `getHostName()` was doing DNS, which caused problems for CallTest.connectTimeoutsAttemptsAlternateRoute().